src/Shared/Recruit.Vacancies.Client/Ioc/ServiceCollectionExtensions.cs

```diff
@@ -116,6 +121,21 @@ private static void RegisterServiceDeps(IServiceCollection services, IConfigurat
             services.AddTransient<ICache, Cache>();
             services.AddTransient<IHtmlSanitizerService, HtmlSanitizerService>();
 
+            //Reporting Service
+            services.AddTransient<ICsvBuilder, CsvBuilder>();
+            services.AddTransient<IReportService, ReportService>();
+            services.AddTransient<ProviderApplicationsReportStrategy>();
+            services.AddTransient<Func<ReportType, IReportStrategy>>(serviceProvider => reportType =>
+            {
+                switch (reportType)
+                {
+                    case ReportType.ProviderApplications:
+                        return serviceProvider.GetService<ProviderApplicationsReportStrategy>();
+                    default:
+                        throw new KeyNotFoundException($"No report strategy for {reportType}");
```

Comment:
Could this just be Exception as I do not see where this is a Key Value lookup.

Reply:
👍